hack/e2e-coverage.sh

```
@@ -0,0 +1,210 @@
+					echo "Error: unsupported OS for Codecov uploader: ${codecov_os}"
+					exit 1
+					;;
+			esac
+			echo "Downloading Codecov uploader for ${codecov_asset}"
+			curl -sS -o "${codecov_bin}" \
+				"https://uploader.codecov.io/${codecov_version}/${codecov_asset}"
+			curl -sS -o "${codecov_bin}.SHA256SUM" \
+				"https://uploader.codecov.io/${codecov_version}/${codecov_asset}.SHA256SUM"
+
+			# macOS ships shasum; Linux ships sha256sum.
+			if !(
```

Shellcheck (0.11.0)

[error] You are missing a required space after the !.

(SC1035)